File: 0001-two-sum/0001-two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    /**
      [2, 7, 11, 15] | 9
      
     */
     const seenMap = new Map();
     for(let i = 0; i < nums.length; i++) {
      const req = target - nums[i];
      if(seenMap.has(req)) {
        return [i, seenMap.get(req)]
      }
      seenMap.set(nums[i], i)
     }
};

File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> indexValueMap = new HashMap<>();
        for(int i = 0; i < nums.length; i++) {
          int rem = target - nums[i];
          if(indexValueMap.containsKey(rem)) {
            return new int[]{i, indexValueMap.get(rem)};
          }
          indexValueMap.put(nums[i], i);
        }
        return new int[]{-1, -1};
    }
}